core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5445,29 +5455,30 @@ targetNamespace, castNonNull(catalogReader.unwrap(Prepare.CatalogReader.class)),
       throw newValidationError(createTable, RESOURCE.createTableRequiresAsQuery());
     }
 
-    final SqlValidatorScope queryScope = requireNonNull(scopes.get(queryNode));
-
     //Note, this can either a row expression or a query expression with an optional ORDER BY
-    if (queryNode instanceof SqlSelect) {
-      final SqlSelect sqlSelect = (SqlSelect) queryNode;
-      validateSelect(sqlSelect, unknownType);
-    } else {
-      validateQuery(queryNode, queryScope, unknownType);
-    }
-
-    final SqlValidatorNamespace queryNS = getNamespaceOrThrow(queryNode);
-    final DdlNamespace createTableNS = (DdlNamespace) getNamespaceOrThrow(createTable);
+    //We're not currently handling the row expression case.
+
+    SqlValidatorScope createTableScope = this.getCreateTableScope(createTable);
+    // In order to be sufficiently general to the input of Create table,
+    // We have to validate this expression in the overall scope of the create table node,
+    // Since the associated query node
+    // doesn't necessarily have to be a select (notably, validaSelect doesn't work if the query has
+    // any 'with' clauses)
+    // Note that we also can't use validateScopedExpression, as this can rewrite the sqlNode
+    // via a call to performUnconditionalRewrites, which should have already been done by the time
+    // that we reach this points, and calling performUnconditionalRewrites twice is likely invalid.
```

Comment:
likely invalid? can you elaborate more

And what do you miss out on by not calling `validateScopedExpression`?

Reply:
`likely invalid? can you elaborate more` : this was mostly stream of thought while I was working through this, having gone back and checked, it seems like it's valid to do the re-write twice, but it's not needed.

Basically, validateScopedExpression does rewriting, does validation, and then possibly does some type derivation. The only thing we need to at this stage is validation (since the rewriting has already occurred), so we just explicitly do the validation. We don't "miss out" on anything, because it's handled in other steps. 

I've updated the comment to be clearer.